src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -5,11 +5,36 @@
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal static partial class BoundExpressionExtensions
     {
+        /// <summary>
+        /// Only legal for expressions that can produce l-values.
```

Comment:
description is too vague. ref ternaries, array accesses, field accesses also can produce lvalues. 

I understand what the helper is for, but it is not what the description says.

Reply:
It seems the helper is legal only for expressions that represent _symbols_ which can have RefKind.

---
In reply to: [153599228](https://github.com/dotnet/roslyn/pull/23415#discussion_r153599228) [](ancestors = 153599228)